macast/server.py

```diff
@@ -200,7 +210,7 @@ def run(self):
             self.protocol.handler.reload()
             cherrypy.engine.publish('ssdp_update_ip')
         # service started
-        cherrypy.engine.block()
+        cherrypy.engine.block(interval=75) # Not really blocking but default 0.1 second timeout
```

Comment:
This change on my computer will cause the app to get stuck when closing the app or stopping the service.

Reply:
That's kind of bad, I didn't get stuck on my windows, it turns out I got too excited, the existence of the default value always makes sense.

Reply:
I got it, because CherryPy uses native win32event objects to block on Windows, the interval argument is ignored.

Reply:
Yes, CherryPy uses time.sleep() to implement block(), and cPython's time.sleep() under windows platform will use WaitForSingleObject() api to respond to SIGINT while waiting a tiemout if called in main thread, POSIX nanosleep or clock_nanosleep used by Linux and MacOS should have the same performance but not only in main thread.  So I'm pretty baffled as to what's causing this problem.

Reply:
https://github.com/cherrypy/cherrypy/blob/5fff5e4b3aea99b051a55f813b7ddd194425eb2d/cherrypy/process/win32.py#L103-L119

Reply:
Thanks for the correction, I found I was looking in the [wrong place](https://github.com/cherrypy/cherrypy/blob/5fff5e4b3aea99b051a55f813b7ddd194425eb2d/cherrypy/process/wspbus.py#L371) 
, this wait() method has been overridden in subclasses already.